### PR TITLE
Temporary fix for /auth/register user profile validation

### PR DESCRIPTION
**Pull request by kbuci**
This is a quick fix that solves the issue in /auth/register, where an API request can be sent to create a user with an invalid profile
The back-end fix involves just passing the profile information to /auth/register and having the server validate it, before a user can be created. This means that the client now has to send the profile data twice, once when calling /auth/register to create the user, and again when calling /users/:id/profile to actually update. 
A more in depth future fix would involve altering front end logic along with back-end to have a single endpoint (such as altering /auth/register) to be responsible for both creating the user and updating the profile and resume, so that the client creates a user and fills in the profile in one transactional request
This approach (and the existing web portal) has the possibility that, in a event of a network issue, a user's application might end up having an empty profile, since creating a account and filling in a profile are separate requests. This PR though requires less front end coordination and CR, and in the meantime will remove the ability to just create a user by only passing an email and password.

### CODE REVIEW

File: app/server/controllers/UserController.js [hunk 1]
```diff
@@ -117,10 +117,26 @@ UserController.loginWithPassword = function(email, password, callback){
...
 
+/**
+ * Creates and validates a new user given an email, password and profile.
+ * @param  {String}   email    User's email.
+ * @param  {String}   password User's password
+ * @param  {Object}   profile  Profile object
+ * @param  {Function} callback args(err, user)
+ */
+UserController.createValidUser = function(email, password, profile, callback){
+  User.validateProfile(profile, function(err){
```
Comment (ryanluu12345) [inline app/server/controllers/UserController.js:128]: Where is this defined? Did this already exist? Or did you add this method for the User model object?

File: app/client/src/services/AuthService.js [hunk 2]
```diff
@@ -73,11 +75,13 @@ angular.module('reg')
...
 
-      authService.createAccount = function(email, password, onSuccess, onFailure) {
+      authService.createAccount = function(email, password, profile, onSuccess, onFailure) {
+        profile.name = profile.firstname + ' ' + profile.lastname;
         return $http
           .post('/auth/register', {
             email: email,
-            password: password
+            password: password,
+            profile: profile
```
Comment (ryanluu12345) [inline app/client/src/services/AuthService.js:84]: By the way, is auth/register set up to handle the profile object? Does it use object deconstruction to get the profile? Or does it pass the entire body to the backend controller?

File: app/client/views/checkin/checkinCtrl.js [hunk 3]
```diff
@@ -182,7 +182,7 @@ angular.module('reg')
       };
 
       function _apply(e){
-        AuthService.createAccount($scope.newUser.email, $scope.newUser.password, function success(user) {
+        AuthService.createAccount($scope.newUser.email, $scope.newUser.password, $scope.newUser.profile, function success(user) {
```
Comment (ryanluu12345) [inline app/client/views/checkin/checkinCtrl.js:185]: Hmm why is there a createAccount call during the checkin portion?

**Review by ryanluu12345 — CHANGES_REQUESTED** [review]

File: app/server/controllers/UserController.js (hunk 1 above)
Comment (kbuci) [reply app/server/controllers/UserController.js:128]: validateProfile already exists in User.js

File: app/client/src/services/AuthService.js (hunk 2 above)
Comment (kbuci) [reply app/client/src/services/AuthService.js:84]: auth/register is set up to take the entire profile object on the back-end

File: app/client/views/checkin/checkinCtrl.js (hunk 3 above)
Comment (kbuci) [reply app/client/views/checkin/checkinCtrl.js:185]: I think this might be for the admin to create users during check-in (for walk-in hackers)

**Pull request closed without merging**